File: employee-management-frontend/src/services/ employeeService.js
import API from "../utils/axiosConfig";

export const getAllEmployees = async () => {

    return await API.get("/hr/employees");
};

export const addEmployee = async (employee) => {

    return await API.post(
        "/hr/employees",
        employee
    );
};

export const getEmployeeById = async (id) => {

    return await API.get(`/employees/${id}`);
};

export const searchEmployeeByName = async (name) => {

    return await API.get(
        `/employees/name/${name}`
    );
};

export const deleteEmployee = async (id) => {

    return await API.delete(
        `/hr/employees/${id}`
    );
};

File: employee-management/src/main/java/com/example/employee_management/service/EmployeeService.java
package com.example.employee_management.service;

import com.example.employee_management.dto.SalaryIncrementDTO;
import com.example.employee_management.entity.Employee;

import java.util.List;

public interface EmployeeService {

    Employee addEmployee(Employee employee);

    Employee updateEmployee(Long id, Employee employee);

    void deleteEmployee(Long id);

    List<Employee> getAllEmployees();

    Employee getEmployeeById(Long id);

    List<Employee> searchEmployeeByName(String name);

    Employee incrementSalary(Long id, SalaryIncrementDTO dto);
}